src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -871,6 +871,24 @@ private static void SetPathAndQueryParameters(CodeMethod target, OpenApiUrlTreeN
     }
     private CodeTypeBase CreateComposedModelDeclaration(OpenApiUrlTreeNode currentNode, OpenApiSchema schema, OpenApiOperation operation, string suffixForInlineSchema, CodeNamespace codeNamespace) {
         var typeName = currentNode.GetClassName(config.StructuredMimeTypes, operation: operation, suffix: suffixForInlineSchema, schema: schema).CleanupSymbolName();
+        var typesCount = schema.AnyOf?.Count ?? schema.OneOf?.Count ?? 0;
+        if ((typesCount == 1 && schema.Nullable && schema.IsAnyOf()) || // nullable on the root schema outside of anyOf
+            typesCount == 2 && schema.AnyOf.Any(static x => // nullable on a schema in the anyOf
+                                                        x.Nullable &&
+                                                        !x.Properties.Any() &&
```

Comment:
```suggestion
                                                        x.IsObject() &&
                                                        !x.Properties.Any() &&
```
Should we validate that the type is an object as well?

Reply:
I had added it originally, but removed it. Technically it can be assumed it's an object (is object only checks for type == "object")

Reply:
Thanks for clarifying that one.